internal/builders/generic/attest.go

```diff
@@ -20,6 +20,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"os"
+	"path"
```

Comment:
why not set a default value for the provenance file in https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/workflows/generator_generic_slsa3.yml#L37, and let users provide the name they want?

Reply:
I think we may run into problems with the secure-upload-artifact Actions. They currently assume files are in the root folder, I think

Reply:
edit: we use the upload-artifact, not the secure one so my last comment is wrong

Reply:
> why not set a default value for the provenance file in https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/workflows/generator_generic_slsa3.yml#L37?

I'd argue against this, if just for the fact that users would have to specify the input when the didn't before so it adds friction to upgrading and folks have complained previously about having to add new inputs.